From 1a383dca2f60707ff9e234f9a56c89eb90aad927 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Wed, 23 Aug 2023 07:35:27 +0200
Subject: [PATCH] php8-pecl-sodium: Backport from OpenWrt 22.03

This package is needed by various Nextcloud plugind, so let's backport
it.
---
 lang/php8-pecl-sodium/Makefile | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 lang/php8-pecl-sodium/Makefile

diff --git a/lang/php8-pecl-sodium/Makefile b/lang/php8-pecl-sodium/Makefile
new file mode 100644
index 000000000..8f8473f04
--- /dev/null
+++ b/lang/php8-pecl-sodium/Makefile
@@ -0,0 +1,33 @@
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PECL_NAME:=sodium
+PECL_LONGNAME:=Bindings for libsodium
+
+PKG_VERSION:=2.0.23
+PKG_RELEASE:=$(AUTORELEASE)
+PKG_HASH:=f8c04533de8e4b48077f65de61b187e195c0919c41d4543e2263604fa0c50379
+
+PKG_NAME:=php8-pecl-sodium
+PKG_SOURCE:=lib$(PECL_NAME)-$(PKG_VERSION).tgz
+PKG_SOURCE_URL:=http://pecl.php.net/get/
+
+PKG_MAINTAINER:=W. Michael Petullo <mike@flyn.org>
+
+PKG_LICENSE:=ISC
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_BUILD_DEPENDS:=php8
+PKG_BUILD_DIR:=$(BUILD_DIR)/pecl-php8/lib$(PECL_NAME)-$(PKG_VERSION)
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/nls.mk
+include ../php8/pecl.mk
+
+$(eval $(call PHP8PECLPackage,sodium,$(PECL_LONGNAME),+libsodium,30))
+$(eval $(call BuildPackage,$(PKG_NAME)))
-- 
2.41.0

